### PR TITLE
fix in employing context-lines feature for all diff/compare commands

### PR DESCRIPTION
**Pull request by ATorrise**
BugFix: Fix in employing `--context-lines` option for all diff/compare commands. Fixed broken `--seqnum` option implementation. Corrected and added unit, system and integration tests

Signed-off-by: Amber Torrise <amber.torrise@broadcom.com>

### CODE REVIEW

**Review by KevinLoesch1 — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
Thanks for fixing this.
LGTM! :yum:

**Review by t1m0thyj — APPROVED** [review]
Should we add unit tests for these commands verifying that the `contextLines` argument behaves as expected?

I'm also ok with merging the PR as-is since all coverage checks are met 🙂

File: packages/cli/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 All notable changes to the Zowe CLI package will be documented in this file.
 
+## Recent Changes
+- BugFix: Fix in employing `--context-lines` option for all diff/compare commands. Fixed broken `--seqnum` option implementation. Corrected and added unit, system and integration tests [#1529](https://github.com/zowe/zowe-cli/issues/1529)
```
Comment (t1m0thyj) [inline packages/cli/CHANGELOG.md:6]: We mention only user-facing changes in the changelog, so no need to mention "Corrected and added unit, system and integration tests" here 🙂

File: packages/cli/__tests__/zosfiles/__integration__/compare/ds/__scripts__/bogus_flag.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+zowe zos-files compare ds "AT895452.TESTING.JCL(ECONTXTL)" "AT895452.TESTING.JCL(CONTXTLN)" --bogus-flag
```
Comment (t1m0thyj) [inline packages/cli/__tests__/zosfiles/__integration__/compare/ds/__scripts__/bogus_flag.sh:3]: Please replace mainframe ID with a generic one like "IBMUSER".

**Review by t1m0thyj — COMMENTED** [review]
Thanks for adding/fixing tests, left a few comments

File: packages/cli/__tests__/zosfiles/__system__/compare/ds/Compare.ds.system.test.ts [hunk 3]
```diff
@@ -22,12 +22,12 @@ let testEnvironment: ITestEnvironment<ITestPropertiesSchema>;
...
 let dsname: string;
 const TIMEOUT = 20000;
-
+const buffer = Buffer.from("buff");
 
 describe("Compare Data Sets", () => {
     beforeAll(async () => {
         testEnvironment = await TestEnvironment.setUp({
-            installPlugin: true,
+            // installPlugin: true,
```
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__system__/compare/ds/Compare.ds.system.test.ts:30]: Is there a reason why we want to keep this here commented?

File: packages/cli/__tests__/zosfiles/__unit__/compare/ds/Dataset.handler.unit.test.ts [hunk 4]
```diff
@@ -108,6 +110,45 @@ describe("Compare data set handler", () => {
...
+                }
+            };
+            const options: IDiffOptions = {
+                contextLinesArg,
+                outputFormat: "terminal"
+            };
+
+            DiffUtils.getDiffString = jest.fn(async () => {
+                return "compared string";
+            });
```
Comment (zFernand0) [inline packages/cli/__tests__/zosfiles/__unit__/compare/ds/Dataset.handler.unit.test.ts:129]: I know that subsequent tests are not looking for any output of the called functions, however please keep in mind that this stile of mocking may result in unexpected test failues.
For example, someone adding tests (after this one) might expect the real behavior of the `getDiffString` method and they will get a mocked version instead.
Suggestion: For future tests, please try using `jest.spyOn` (if possible) or `jest.mock` when dealing with completed modules 😋

**Review by zFernand0 — COMMENTED** [review]
The code changes and tests LGTM!
I do agree with Timothy's comment on the changelog and the sh script.
Happy to approve once those are addressed.

File: packages/cli/__tests__/zosfiles/__unit__/compare/ds/Dataset.handler.unit.test.ts (hunk 4 above)
Comment (ATorrise) [reply packages/cli/__tests__/zosfiles/__unit__/compare/ds/Dataset.handler.unit.test.ts]: I've added this mocked method approach

File: packages/cli/src/zosfiles/compare/lf-sdd/LocalfileSpooldd.handler.ts [hunk 5]
```diff
@@ -59,22 +59,21 @@
~            }
~
~            // reading local file as buffer
~            lfContentBuf = fs.readFileSync(localFileHandle);
~        } finally {
             fs.closeSync(localFileHandle);
         }
 
+        const lfContentBuf = fs.readFileSync(localFile);
```
[the remark this answers is not in the document]
Comment (ATorrise) [reply packages/cli/src/zosfiles/compare/lf-sdd/LocalfileSpooldd.handler.ts]: whoops

File: packages/cli/src/zosfiles/compare/lf-sdd/LocalfileSpooldd.handler.ts [hunk 6]
```diff
@@ -59,22 +59,21 @@
~            }
~
~            // reading local file as buffer
~            lfContentBuf = fs.readFileSync(localFileHandle);
~        } finally {
             fs.closeSync(localFileHandle);
         }
 
+        lfContentBuf = fs.readFileSync(localFile);
```
[the remark this answers is not in the document]
Comment (ATorrise) [reply packages/cli/src/zosfiles/compare/lf-sdd/LocalfileSpooldd.handler.ts]: whoops pt 2

**Review by t1m0thyj — APPROVED** [review]
Thanks for fixing both the `--context-lines` and `--seqnum` options, and increasing coverage and removing snapshots in the process 🙂

**Review by zFernand0 — APPROVED** [review]
LGTM

**Pull request merged**